ogc/src/main/scala/geotrellis/server/ogc/OgcSource.scala

```diff
@@ -75,14 +69,73 @@ case class SimpleSource(
     CapabilitiesView.boundingBox(crs, reprojected.extent, reprojected.cellSize)
   }
 
-  lazy val time: Option[ZonedDateTime]     = attributes.get("time").map(ZonedDateTime.parse)
   lazy val nativeRE: GridExtent[Long]      = source.gridExtent
   lazy val nativeCrs: Set[CRS]             = Set(source.crs)
   lazy val nativeExtent: Extent            = source.extent
   lazy val metadata: RasterMetadata        = source.metadata
   lazy val attributes: Map[String, String] = metadata.attributes
 }
 
+/**
+ * An imagery source with a [[RasterSource]] that defines its capacities
+ */
+case class SimpleSource(
+  name: String,
+  title: String,
+  source: RasterSource,
+  defaultStyle: Option[String],
+  styles: List[OgcStyle]
+) extends RasterOgcSource {
+  val timeInterval: Option[OgcTimeInterval] = None
+}
+
+case class GeoTrellisOgcSource(
+  name: String,
+  title: String,
+  sourceUri: String,
+  defaultStyle: Option[String],
+  styles: List[OgcStyle],
+  timeMetadataKey: String = "times"
+) extends RasterOgcSource {
+
+  private val dataPath = GeoTrellisPath.parse(sourceUri)
+
+  // If layer is temporally enabled, default time is the first in the layer
+  lazy val source =
+    GeoTrellisRasterSourceLegacy(dataPath, timeInterval.map(_.start))
+
+  lazy val isTemporal: Boolean = !times.isEmpty
+
+  lazy val times: List[ZonedDateTime] = {
+    val layerId = source.dataPath.layerId
+    val header = source.attributeStore.readHeader[LayerHeader](layerId)
+    if (header.keyClass.contains("SpaceTimeKey")) {
+      source.attributeStore.read[List[ZonedDateTime]](layerId, timeMetadataKey).sorted
```

Comment:
Add an import `geotrellis.store.query.CirceCompat._` in all places you use temporal circe codecs. The sotry behind that is that Scala 2.12 circe handles temporal codecs in a differnet way (they are supproted by default) and in early circe versions such codecs lived in a separate circe module.

Reply:
@CloudNiner ^ this is a comment that explains why Scala 2.11 builds are failing.

Reply:
Ok, [here is the problem](https://github.com/locationtech/geotrellis/blob/0e60b860825d289f7283bea9683f3ca4b2f3c4c5/store/src/main/scala/geotrellis/store/json/Implicits.scala#L95-L100) we need to remove it from the GeoTrellis core.

Reply:
Issue here resolved by removing the CirceCompat import from GeoTrellisRasterSourceLegacy which pulls a ZonedDateTime decoder from the package listed in the link above.